pages/docs/couple-your-code/couple-your-code-just-in-time-mapping.md

```diff
@@ -0,0 +1,134 @@
+    <read-data name="Velocities" mesh="ReceivedMeshName" />
+    <write-data name="Forces" mesh="ReceivedMeshName" />
+    <!-- define the just-in-time mapping (read-consistent) to read the velocities, note the empty "to" mesh for the read direction-->
+    <mapping:nearest-neighbor direction="read" from="ReceivedMeshName" constraint="consistent" />
+    <!-- define the just-in-time mapping (write conservative) to write forces just-in-time, note the empty "from" mesh for the write direction-->
+    <mapping:nearest-neighbor direction="write" to="ReceivedMeshName" constraint="conservative" />
+  </participant>
+```
+
+Here, "SolverOne" now defines a just-in-time mapping access in read-direction for data "Velocities" and in write direction for data "Forces". In its core, using just-in-time mapping requires three changes in the configuration file:
+
+1. `api-access` needs to be enabled on the `receive-mesh` (similar to the direct access), "ReceivedMeshName" in our example above.
+2. The `read-data` and `write-data` tags need to use the received mesh with api-access enabled.
+3. The mapping only defines (depending on the direction) a "from" or a "to" mesh. For read-consistent mappings, only the "from" mesh must be defined. For write-conservative mappings, only the "to" mesh must be defined. In both cases, the defined mesh needs to match the received mesh with api-access enabled (which is also used in the `read-data` or `write-data` tag).
+
+At its core, the API now makes use of two new API functions, called `mapAndReadData` and `mapAndWriteData`. Both functions are very similar to the analogous functions `readData` and `writeData`: the only difference is that the functions take spatial coordinates as function arguments instead of static vertexIDs stemming from the preCICE initialization. In addition to `mapAndReadData` and `mapAndWriteData`, the access region needs to be defined by "SolverOne" using `setMeshAccessRegion`. The full code example reads the following:
+
+```cpp
+    // Note that "ReceivedMeshName" is defined and received
+    // from another participant
+    const std::string otherMesh = "ReceivedMeshName";
+
+    // Allocate and fill a 'boundingBox' to define an access region
+    // in our example, we use the unit cube.
+    // Assuming dim == 3, means that the bounding box has
+    // dim * 2 == 6 elements.
+    std::vector<double> boundingBox {
+        0, 0, 0, // minimum corner
+        1, 1, 1 // maximum corner
+    };
+
+    // Define region of interest, where we want to obtain the direct access.
+    // See also the API documentation of this function for further notes.
+    precice.setMeshAccessRegion(otherMesh, boundingBox);
+
+    // initialize preCICE as usual
+    // initializing data (requiresInitialData) before calling initialize is not possible, as we first need to exchange the
+    // meshes before being able to access the mesh
```

Comment:
What does "not possible" exactly mean? Return false? Throws an error?
What if other data (e.g. in a different coupling scheme) requires initialization?

Reply:
"not possible" means that it is not possible by construction.

Usually, we have 

```
requiresInitialData() -> initialize() -> getMeshVertexIDsAndCoordinates()
```
where the last function gives us only the IDs we need for `writeData` inside the `requiresInitialData()`, i.e., the above is not possible at all, as we simply don't have the vertex IDs where we need them (inside the `requiresInitialData()`.